File: src/main/java/io/github/binaryfoo/decoders/CardholderVerificationConditionCode.java
package io.github.binaryfoo.decoders;

/**
 * When a given method for verifying a cardholder should be applied.
 */
public enum CardholderVerificationConditionCode {
    Always(0x00, "Always"),
    UnattendedCash(0x01, "If unattended cash"),
    NotStuff(0x02, "If not (unattended cash, manual cash, purchase + cash)"),
    TerminalSupports(0x03, "If terminal supports CVM"),
    ManualCash(0x04, "If manual cash"),
    PurchasePlusCash(0x05, "If purchase + cash"),
    TxLessThanX(0x06, "If transaction in application currency and < X"),
    TxMoreThanX(0x07, "If transaction in application currency and >= X"),
    TxLessThanY(0x08, "If transaction in application currency and < Y"),
    TxMoreThanY(0x09, "If transaction in application currency and >= Y");

    private final int code;
    private final String description;

    CardholderVerificationConditionCode(int code, String description) {
        this.code = code;
        this.description = description;
    }

    public String getDescription() {
        return description;
    }

    public static CardholderVerificationConditionCode fromCode(int code) {
        for (CardholderVerificationConditionCode conditionCode : CardholderVerificationConditionCode.values()) {
            if (conditionCode.code == code) {
                return conditionCode;
            }
        }
        return null;
    }
}

File: src/main/java/io/github/binaryfoo/decoders/CardholderVerificationConditionCode.kt
package io.github.binaryfoo.decoders

/**
 * When a given method for verifying a cardholder should be applied.
 */
enum class CardholderVerificationConditionCode(val code: Int, val description: String) {
  Always(0, "Always"),
  UnattendedCash(1, "If unattended cash"),
  NotStuff(2, "If not (unattended cash, manual cash, purchase + cash)"),
  TerminalSupports(3, "If terminal supports CVM"),
  ManualCash(4, "If manual cash"),
  PurchasePlusCash(5, "If purchase + cash"),
  TxLessThanX(6, "If transaction in application currency and < X"),
  TxMoreThanX(7, "If transaction in application currency and >= X"),
  TxLessThanY(8, "If transaction in application currency and < Y"),
  TxMoreThanY(9, "If transaction in application currency and >= Y");

  companion object {
    fun fromCode(code: Int): CardholderVerificationConditionCode? = values().firstOrNull { it.code == code }
  }
}
